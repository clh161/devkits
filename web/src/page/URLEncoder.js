// @flow strict
import type { Node } from 'react';
import React, { useState } from 'react';
import { Divider, Grid, Typography } from '@material-ui/core';
import { Helmet } from 'react-helmet';

const DEFAULT_URL =
  'https://devkits.net/json=%7B%0A%20%20%22user%22:%20%7B%0A%20%20%20%20%22id%22:%201,%0A%20%20%20%20%22name%22:%20%22devkits%22%0A%20%20%7D%0A%7D%0A';
const DECODED_URL = decodeURI(DEFAULT_URL);
export default function URLEncoder(): Node {
  const [encodedURL, setEncodedURL] = useState<string>(DEFAULT_URL);
  const [decodedURL, setDecodedURL] = useState<string>(DECODED_URL);

  function onEncodedURLChange(event) {
    const { value } = event.target;
    setEncodedURL(value);
    try {
      setDecodedURL(decodeURI(value));
    } catch (_) {
      setDecodedURL(value);
    }
  }

  function onDecodedURLChange(event) {
    const { value } = event.target;
    setDecodedURL(value);
    setEncodedURL(encodeURI(value));
  }

  return (
    <div>
      <Helmet>
        <title>URL Encoder</title>
        <meta content="Encode or decode URL." name="description" />
        <meta
          content="URL Encoder, URL Decoder, encoding URL, decoding URL"
          name="keywords"
        />
      </Helmet>
      <Grid container spacing={3}>
        <Grid item xs={12}>
          <Typography component="h1" variant="h4">
            URL Encoder
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <textarea
            onChange={onEncodedURLChange}
            placeholder="Encoded URL"
            style={{ width: '100%', minHeight: 160 }}
            value={encodedURL}
          />
        </Grid>
        <Divider />
        <Grid item xs={12}>
          <textarea
            onChange={onDecodedURLChange}
            placeholder="Decoded URL"
            style={{ width: '100%', minHeight: 160 }}
            value={decodedURL}
          />
        </Grid>
      </Grid>
    </div>
  );
}
