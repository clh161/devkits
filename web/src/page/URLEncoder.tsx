import { Divider, Grid, TextField } from '@mui/material';
import React, { ReactElement, useState } from 'react';
const DEFAULT_URL =
  'https://devkits.net/json=%7B%0A%20%20%22user%22:%20%7B%0A%20%20%20%20%22id%22:%201,%0A%20%20%20%20%22name%22:%20%22devkits%22%0A%20%20%7D%0A%7D%0A';
const DECODED_URL = decodeURIComponent(DEFAULT_URL);
export default function URLEncoder(): ReactElement {
  const [encodedURL, setEncodedURL] = useState<string>(DEFAULT_URL);
  const [decodedURL, setDecodedURL] = useState<string>(DECODED_URL);

  function onEncodedURLChange(event) {
    const { value } = event.target;
    setEncodedURL(value);

    try {
      setDecodedURL(decodeURIComponent(value));
    } catch (_) {
      setDecodedURL(value);
    }
  }

  function onDecodedURLChange(event) {
    const { value } = event.target;
    setDecodedURL(value);
    setEncodedURL(encodeURIComponent(value));
  }

  return (
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <TextField
          label='Encoded URL'
          maxRows={15}
          minRows={5}
          multiline
          onChange={onEncodedURLChange}
          placeholder='Encoded URL'
          style={{
            width: '100%',
          }}
          value={encodedURL}
        />
      </Grid>
      <Divider />
      <Grid item xs={12}>
        <TextField
          label='Decoded URL'
          maxRows={15}
          minRows={5}
          multiline
          onChange={onDecodedURLChange}
          placeholder='Decoded URL'
          style={{
            width: '100%',
          }}
          value={decodedURL}
        />
      </Grid>
    </Grid>
  );
}
