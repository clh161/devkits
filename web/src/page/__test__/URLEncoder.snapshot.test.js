// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import URLEncoder from '../URLEncoder';

it('Init with default', () => {
  const tree = renderer.create(<URLEncoder />).toJSON();
  expect(tree).toMatchSnapshot();
});
