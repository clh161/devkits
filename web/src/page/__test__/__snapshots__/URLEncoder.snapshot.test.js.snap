// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init with default 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h1
        className="MuiTypography-root MuiTypography-h4"
      >
        URL Encoder
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <textarea
        onChange={[Function]}
        placeholder="Encoded URL"
        style={
          Object {
            "minHeight": 160,
            "width": "100%",
          }
        }
        value="https://devkits.net/json=%7B%0A%20%20%22user%22:%20%7B%0A%20%20%20%20%22id%22:%201,%0A%20%20%20%20%22name%22:%20%22devkits%22%0A%20%20%7D%0A%7D%0A"
      />
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <textarea
        onChange={[Function]}
        placeholder="Decoded URL"
        style={
          Object {
            "minHeight": 160,
            "width": "100%",
          }
        }
        value="https://devkits.net/json={
  \\"user\\": {
    \\"id\\": 1,
    \\"name\\": \\"devkits\\"
  }
}
"
      />
    </div>
  </div>
</div>
`;

exports[`Init with default 2`] = `
Object {
  "baseTag": Array [],
  "bodyAttributes": Object {},
  "defer": true,
  "encode": true,
  "htmlAttributes": Object {},
  "linkTags": Array [],
  "metaTags": Array [
    Object {
      "content": "Encode or decode URL.",
      "name": "description",
    },
    Object {
      "content": "URL Encoder, URL Decoder, encoding URL, decoding URL",
      "name": "keywords",
    },
  ],
  "noscriptTags": Array [],
  "onChangeClientState": [Function],
  "scriptTags": Array [],
  "styleTags": Array [],
  "title": "URL Encoder",
  "titleAttributes": Object {},
}
`;
